trie/trie.go

```diff
@@ -405,6 +405,10 @@ func (t *Trie) delete(n node, prefix, key []byte) (bool, node, error) {
 		n.flags = t.newFlag()
 		n.Children[key[0]] = nn
 
+		if nn != nil {
+			// n still contains at least two values and cannot be reduced.
+			return true, n, nil
+		}
```

Comment:
@Evolution404 can you explain a bit how this optimization works exactly?

Reply:
Since `n` is a fullNode, there must be at least two children.  In other word, no matter if `nn` is `nil`, another child of `n` should not be `nil`. When `nn` isn't `nil`, these two children make `n` still be a `fullNode` . There is no need to the following check.

Reply:
@fjl

Reply:
![1](https://user-images.githubusercontent.com/35091674/122558510-bfefa400-d070-11eb-9493-17d6957c4d7e.png)

Reply:
Thanks!

Reply:
I have added this info to the comment.